src/sotauptaneclient.cc

```diff
@@ -21,8 +21,8 @@ void SotaUptaneClient::run(command::Channel *commands_channel) {
 }
 
 bool SotaUptaneClient::isInstalled(const Uptane::Target &target) {
-  if (target.ecu_identifier() == config.uptane.primary_ecu_serial) {
-    return target.filename() == OstreePackage::getEcu(config.uptane.primary_ecu_serial, config.ostree.sysroot).ref_name;
+  if (target.ecu_identifier() == uptane_repo.primary_ecu_serial) {
+    return target.filename() == OstreePackage::getEcu(uptane_repo.primary_ecu_serial, config.ostree.sysroot).ref_name;
```

Comment:
There is a straightforward solution of having a getter for primary_ecu_serial, but this will not make it any less tangled. The problem is that we need primary_ecu_serial in both SotaUptaneClient (for finding targets that should be installed on primary) and in Uptane::Repository (because primary ecu is a part of UPTANE specification and should be managed accordingly). It can also be generated dynamically which is done in the initialization procedure, which in turn I had to place in Uptane::Repository, because it uses http connection and secondaries. So a "true" solution could be to move the main initialization sequence to SotaUptaneClient, while keeping only certain subfunctions in Uptane::Repository. We could then generate primary_ecu_serial in the client and just pass it as a constructor parameter to Uptane::Repository.